src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/executors/spark/SparkExecutor.scala

```diff
@@ -51,9 +52,14 @@ object SparkExecutor extends Executor {
     val id = randomUUID().toString
     val ji = jobInstance.copy(executorJobId = Some(id), jobStatus = Submitting)
     updateJob(ji).map { _ =>
-      val running = getSparkLauncher(id, ji.jobName, ji.jobParameters).launch()
-      Thread.sleep(SparkExecutorConfig.getSubmitTimeOut)
-      running.destroyForcibly()
+      val start = System.currentTimeMillis()
+      val submitTimeOut = SparkExecutorConfig.getSubmitTimeOut
+      val sparkAppHandle = getSparkLauncher(id, ji.jobName, ji.jobParameters).startApplication()
+      while(
```

Comment:
I think this logic can be written in an event-driven way with the `SparkAppHandle.Listener` and a `CountDownLatch` with a timeout

Reply:
Thanks done